### PR TITLE
Add data-loaders requirement to Mate api guide

### PR DESCRIPTION
**Pull request by VinnieJ-2k20**
Inspired by true events

### CODE REVIEW

File: mate-api.md [hunk 1]
```diff
@@ -214,23 +214,59 @@ class UserRepository {
 }
 ```
 
-  
 
-#### 1.4. Store common entity related errors in constants
 
-  
+#### 1.4. Use data-loaders instead of ORM methods in `find` and `get` methods
```
Comment (vadim-ilchenko) [inline mate-api.md:220]: add to headings

File: mate-api.md [hunk 2]
```diff
@@ -214,23 +214,59 @@ class UserRepository {
...
+
+
+
+>❓Why? If the record is already fetched, it's better to use it instead of fetching it again. Data-loaders are used to cache the fetched data and return it if it's already fetched. Also, lots of separate DB calls are merged into one batch call in data-loader case. See examples of data-loaders in *.loaders folders in api modules.
```
Comment (vadim-ilchenko) [inline mate-api.md]: you can add link to the library

**Review by vadim-ilchenko — APPROVED** [review]

File: mate-api.md (hunk 2 above)
Comment (VinnieJ-2k20) [reply mate-api.md]: Done

File: mate-api.md (hunk 1 above)
Comment (VinnieJ-2k20) [reply mate-api.md:220]: Done

**Pull request merged**